src/schemes/fluid/surface_normal_sph.jl

```diff
@@ -0,0 +1,170 @@
+struct ColorfieldSurfaceNormal{ELTYPE, K}
+    smoothing_kernel::K
+    smoothing_length::ELTYPE
+end
+
+function ColorfieldSurfaceNormal(; smoothing_kernel, smoothing_length)
+    return ColorfieldSurfaceNormal(smoothing_kernel, smoothing_length)
+end
+
+function create_cache_surface_normal(surface_normal_method, ELTYPE, NDIMS, nparticles)
+    return (;)
+end
+
+function create_cache_surface_normal(::ColorfieldSurfaceNormal, ELTYPE, NDIMS, nparticles)
+    surface_normal = Array{ELTYPE, 2}(undef, NDIMS, nparticles)
+    neighbor_count = Array{ELTYPE, 1}(undef, nparticles)
+    return (; surface_normal, neighbor_count)
+end
+
+@inline function surface_normal(particle_system::FluidSystem, particle)
+    (; cache) = particle_system
+    return extract_svector(cache.surface_normal, particle_system, particle)
+end
+
+# Section 2.2 in Akinci et al. 2013 "Versatile Surface Tension and Adhesion for SPH Fluids"
+# Note: This is the simplest form of normal approximation commonly used in SPH and comes
+# with serious deficits in accuracy especially at corners, small neighborhoods and boundaries
+function calc_normal_akinci!(system, neighbor_system::FluidSystem, u_system, v,
+                             v_neighbor_system, u_neighbor_system, semi, surfn)
+    (; cache) = system
+    (; smoothing_kernel, smoothing_length) = surfn
+
+    system_coords = current_coordinates(u_system, system)
+    neighbor_system_coords = current_coordinates(u_neighbor_system, neighbor_system)
+    nhs = get_neighborhood_search(system, neighbor_system, semi)
+
+    if smoothing_length != system.smoothing_length ||
+       smoothing_kernel !== system.smoothing_kernel
+        # TODO: this is really slow but there is no way to easily implement multiple search radia
+        search_radius = compact_support(smoothing_kernel, smoothing_length)
+        nhs = PointNeighbors.copy_neighborhood_search(nhs, search_radius,
+                                                      nparticles(system))
+        PointNeighbors.initialize!(nhs, system_coords, neighbor_system_coords)
+    end
```

Comment:
Do we want to support this with terrible performance, or rather not support it at all (yet)?

Reply:
I think this should be in here. Otherwise this just gets forgotten when this can be done more efficiently.

Reply:
Then please add an issue, so we don't forget.

Reply:
#668